### PR TITLE
Update on spectral invariants and name optimization

### LINKED ISSUE #398: Update on spectral invariants
- [x] Remove the invariant **"Smallest L eigenvalue"**, This invariant is unnecessary because this number is always zero.
- [x] Implement numeric spectral Invariantes: **Rank** of matrix S, DL, DQ, E.

### PR DESCRIPTION
**Pull request by atilaajones**
Closes #398

### CODE REVIEW

**Review by rosaigor138 — APPROVED** [review]

File: source/store/operations_and_invariants/bool_invariants.py [hunk 1]
```diff
@@ -342,6 +316,32 @@ def print(graph, precision):
         return Utils.print_boolean(SomeEigenIntegerDQ.calculate(graph), precision)
 
 
+class SomeEigenIntegerS(InvariantBool):
+    name = "Some S-eigenvalue integer"
```
Comment (rosaigor138) [inline source/store/operations_and_invariants/bool_invariants.py:320]: The name of some-eigenValue is not changed to some-eigen here either, is this correct?

Comment (atilaajones) [reply source/store/operations_and_invariants/bool_invariants.py:320]: Its Correct. Because the lenght is ok.

**Comment by fsoupimenta** [comment]
@atilaajones can you resolve the conflicts before i do my review?

**Comment by atilaajones** [comment]
@fsoupimenta @rosaigor138 
The conflicts have been resolved. Please check that the length of the invariant names is correct.

**Review by fsoupimenta — APPROVED** [review]

**Pull request merged**